src/ParquetMsg.chpl

```diff
@@ -375,18 +400,56 @@ module ParquetMsg {
     }
   }
 
-  proc processParquetFilenames(filenames: [] string, matchingFilenames: [] string) throws {
+  proc processParquetFilenames(filenames: [] string, matchingFilenames: [] string, mode: int) throws {
     var warnFlag: bool;
-    if matchingFilenames.size > 0 {
-      warnFlag = true;
+    if mode == APPEND {
+      var allexist = true;
+      var anyexist = false;
+      for f in filenames {
+        var result =  try! exists(f);
+        allexist &= result;
+        if result {
+          anyexist = true;
+        }
+      }
+      if !anyexist {
+              throw getErrorWithContext(
+                 msg="Cannot append a non-existent file, please save without mode='append'",
+                 lineNumber=getLineNumber(), 
+                 routineName=getRoutineName(), 
+                 moduleName=getModuleName(), 
+                 errorClass='WriteModeError'
+                                        );
+      }
+      if !allexist || (matchingFilenames.size != filenames.size) {
+        throw getErrorWithContext(
+                   msg="appending to existing files must be done with the same number " +
+                      "of locales. Try saving with a different directory or filename prefix?",
+                   lineNumber=getLineNumber(), 
+                   routineName=getRoutineName(), 
+                   moduleName=getModuleName(), 
+                   errorClass='MismatchedAppendError'
+              );
+      }
```

Comment:
I may be missing something here, but it appears that we are duplicating our checks for file existence. I am not sure the `exists()` check on the file names is needed. It appears redundant to me with 
```python
var matchingFilenames = glob("%s_LOCALE*%s".format(filename, ".parquet"));

var warnFlag = processParquetFilenames(filenames, matchingFilenames, mode);
```
being called. 
Is there a reason that we cannot just check 
```python
if matchingFilenames.size == 0{
    # throw error
}
if matchingFilename.size != filenames.size {
    # throw error
}
```

Reply:
I believe the reason that we do two similar checks that would both fail in the same case is that the two cases are different and so we want to generate a different error message.

If there are no files, we let the user know we haven't found any, if there aren't the correct number of files, we let the user know that. So they both would fail under the same conditions, but presenting a different error message in both cases is helpful.

Note that this is drawn from the HDF5 code, so a change here would likely mean we want to change the HDF5 code as well.

Reply:
My two cents... make it match the HDF5 code then file an issue, we should also look into unifying the duplicated code.

Reply:
@mhmerrill - sounds good. Issue #1229 submitted.